backend/src/modules/idim-webservice/idim-webservice.service.ts

```diff
@@ -132,4 +142,109 @@ export class IdimWebserviceService {
             );
         }
     }
+
+    async verifyBceidUser(
+        userId: string,
+        requesterUserGuid: string,
+        requesterAccountTypeCode: string
+    ): Promise<HttpException | BCEIDUserResponse> {
+        if (!this.hasAllRequiredParams()) {
+            return new HttpException(
+                {
+                    error: 'Missing IDIM web service crednetials in the environment variables',
+                },
+                HttpStatus.INTERNAL_SERVER_ERROR
+            );
+        }
+
+        try {
+            const client = await this.getSoapClient();
+            // set xml schema parameters for the request
+            const requestData = {
+                accountDetailRequest: {
+                    onlineServiceId: this.idimWebServiceID,
+                    // who is sending the request
+                    requesterAccountTypeCode,
+                    requesterUserGuid,
+                    // who we search for, exact match userID
+                    userId,
+                    accountTypeCode: RequesterAccountTypeCode.Business,
+                },
+            };
+
+            return new Promise((resolve, reject) => {
+                client.BCeIDService.BCeIDServiceSoap.getAccountDetail(
+                    requestData,
+                    function (error, foundUser) {
+                        // this will be any error from the IDIM server side
+                        if (error) {
+                            return reject(
+                                new HttpException(
+                                    {
+                                        error:
+                                            'IDIM web service call error: ' +
+                                            error,
+                                    },
+                                    HttpStatus.INTERNAL_SERVER_ERROR
+                                )
+                            );
+                        }
+
+                        if (
+                            foundUser.getAccountDetailResult.code == 'Failed' &&
+                            foundUser.getAccountDetailResult.failureCode ==
+                                'NoResults'
+                        ) {
+                            const response = new BCEIDUserResponse();
+                            response.found = false;
+                            response.userId = userId;
+                            return resolve(response);
+                        }
```

Comment:
Hi Catherine, I might not be knowing the context of their soap response result.... the code above and code below are a bit confusing.
When the soap returns "Failed", why do we treat them differently?
If this handling is correct, could we add this special handling in brief at the "method comment" at the beginning? Thank you.

Reply:
Got it.